src/OpenTelemetry.Api/Trace/ActivityExtensions.cs

```diff
@@ -80,6 +80,8 @@ public static Status GetStatus(this Activity activity)
         /// </summary>
         /// <param name="activity">Activity instance.</param>
         /// <param name="kind">Activity execution kind.</param>
+        /// <remarks>This extension method should only be used on <see cref="Activity"/> instances that were created
```

Comment:
Seems likely to confuse people. What if we moved it into ActivitySourceAdapter and made it private?

Reply:
if we do that, we will have to share this between all projects that uses. And, people that need to create an instrumentation/exporter would have to create something like this. The idea (i think) was to facilitate people in that scenario.

Reply:
Going off of memory here, but the only time we need to set the ActivityKind is for legacy Activity objects. We pass those to ActivitySourceAdapter. So the idea is, instead of setting the Kind and then calling ActivitySourceAdapter, just pass Kind to ActivitySourceAdapter and let it do it via the private method. 🤷

Reply:
@CodeBlanch Agree to make it part of Adapter only. There should be no other place where this is required.